### PR TITLE
Return specific error code.

### PR DESCRIPTION
**Pull request by zzl0**
@mckelvin pls review

### CODE REVIEW

File: src/ConnectionPool.cpp [hunk 1]
```diff
@@ -380,17 +380,19 @@ int ConnectionPool::waitPoll() {
...
 
-  int ret_code = 0;
+  err_code_t ret_code = OK_ERR;
   while (m_nActiveConn) {
     int rv = poll(pollfds, n_fds, s_pollTimeout);
     if (rv == -1) {
       markDeadAll(pollfds, keywords::kPOLL_ERROR);
-      return -1;
+      ret_code = POLL_ERR;
+      break;
```
Comment (zzl0) [inline src/ConnectionPool.cpp]: I want to minimize `return` statement.

File: src/ConnectionPool.cpp [hunk 2]
```diff
@@ -449,16 +451,19 @@ int ConnectionPool::waitPoll() {
...
               goto next_fd;
               break;
             case MC_SERVER_ERR:
               markDeadConn(conn, keywords::kSERVER_ERROR, pollfd_ptr);
-              ret_code = -1;
+              ret_code = MC_SERVER_ERR;
               m_nActiveConn -= 1;
               goto next_fd;
               break;
+            default:
```
Comment (zzl0) [inline src/ConnectionPool.cpp]: `default` is added to fix warnnings.

**Comment by zzl0** [comment]
@lembacon pls review

File: src/ConnectionPool.cpp [hunk 3]
```diff
@@ -449,16 +451,19 @@ int ConnectionPool::waitPoll() {
...
+              log_warn("should not be here!");
```
Comment (mckelvin) [inline src/ConnectionPool.cpp]: `NOT_REACHED()`  (defined in `include/Common.h`) is preferred.

Comment (zzl0) [inline src/ConnectionPool.cpp]: fixed, thanks

File: libmc/_client.pyx [hunk 4]
```diff
@@ -48,6 +48,17 @@ cdef extern from "Common.h" namespace "douban::mc":
         OPT_HASH_FNV1A_32
         OPT_HASH_CRC_32
 
+    ctypedef enum err_code_t:
+        SEND_ERR
+        RECV_ERR
+        CONN_POLL_ERR
+        POLL_TIMEOUT_ERR
+        POLL_ERR
```
Comment (mckelvin) [inline libmc/_client.pyx]: @zzl0 

```
/Users/kelvin/GitHub/douban/libmc/include/Common.h:186:3: error: expected identifier
  POLL_ERR = -4,
  ^
/usr/include/sys/signal.h:263:18: note: expanded from macro 'POLL_ERR'
#define POLL_ERR        4       /* [XSR] I/O error */
```

consider adding a unified prefix (e.g.; `MC_RETURN_`? ) to all enums of `err_code_t` type?

**Comment by zzl0** [comment]
@mckelvin done. Tests passed on Mac

**Comment by mckelvin** [comment]
lgtm :beer:

**Pull request merged**